CRM/Contact/BAO/Query.php

```diff
@@ -5054,7 +5054,12 @@ public function generatePermissionClause($onlyDeleted = FALSE, $count = FALSE) {
 
       if (CRM_Core_Permission::check('access deleted contacts')) {
         if (!$onlyDeleted) {
-          $this->_permissionWhereClause = str_replace('( 1 )', '(contact_a.is_deleted = 0)', $this->_permissionWhereClause);
+          if ($this->_permissionWhereClause === '( 1 )') {
+            $this->_permissionWhereClause = str_replace('( 1 )', '(contact_a.is_deleted = 0)', $this->_permissionWhereClause);
+          }
+          else {
+            $this->_permissionWhereClause .= ' AND (contact_a.is_deleted = 0)';
```

Comment:
um... in fact why wouldn't this produce exactly the same result if $this->_permissionWhereClause DID === '( 1 )' ?  I mean, doesn't the SQL become:

`WHERE ( 1 ) AND (contact_a.is_deleted = 0)` ?

Reply:
I seem to recall that being confusing I think and also this is consistent with the code lower down and I would argue that such a change should probably be done in a broader cleanup rather than in a regression fix